### PR TITLE
openvpn3: Fix state directory

### PR DESCRIPTION
**Pull request by name-snrl**
###### Description of changes

This will fix commands such as `openvpn3-admin init-config` and `openvpn3-admin netcfg-service`. Also, for system with `services.resolved.enable = true` the `systemd-resolved` parameter will be added to the configuration file.

I am not sure of the correct way I have chosen to solve this problem, help me do it right.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by name-snrl** [comment]
@KFearsoff @SuperSandro2000 @ckiee @j4m3s-s Could you review, please?

**Review by KFearsoff — APPROVED** [review]
LGMT, thanks for the contribution!

**Review by KFearsoff — APPROVED** [review]

File: nixos/modules/programs/openvpn3.nix [hunk 1]
```diff
@@ -25,6 +25,31 @@ in
       gid = config.ids.gids.openvpn;
     };
 
+    systemd.tmpfiles.rules = [
+      "d /var/lib/openvpn3 0750 openvpn openvpn - -"
+    ];
+
+    systemd.services.openvpn3-netcfg-setup = {
```
Comment (SuperSandro2000) [inline nixos/modules/programs/openvpn3.nix:32]: Shouldn't this run before openvpn3 is started?

File: nixos/modules/programs/openvpn3.nix [hunk 2]
```diff
@@ -25,6 +25,31 @@ in
...
+      description = "Generate openvpn3 configuration";
+      wantedBy = [ "multi-user.target" ];
+
+      serviceConfig.Type = "oneshot";
+      script = with pkgs; ''
+        ${openvpn3}/bin/openvpn3-admin init-config --write-configs
```
Comment (SuperSandro2000) [inline nixos/modules/programs/openvpn3.nix:38]: We should add a package option and use openvpn3 from that

File: nixos/modules/programs/openvpn3.nix [hunk 3]
```diff
@@ -25,6 +25,31 @@ in
...
+        ${if config.services.resolved.enable
+        then ''
+          ${openvpn3}/bin/openvpn3-admin netcfg-service --config-unset resolv-conf
+          ${openvpn3}/bin/openvpn3-admin netcfg-service --config-set systemd-resolved yes
+        ''
+        else ''
+          ${openvpn3}/bin/openvpn3-admin netcfg-service --config-unset systemd-resolved
+          ${openvpn3}/bin/openvpn3-admin netcfg-service --config-set resolv-conf /etc/resolv.conf
+        ''}
```
Comment (SuperSandro2000) [inline nixos/modules/programs/openvpn3.nix:47]: Not sure if making the one argument conditional is cleaner

File: nixos/modules/programs/openvpn3.nix [hunk 4]
```diff
@@ -25,6 +25,31 @@ in
...
+        service="$(busctl status net.openvpn.v3.netcfg | grep Unit=dbus | cut -d= -f2)"
+        systemctl restart "$service"
```
Comment (SuperSandro2000) [inline nixos/modules/programs/openvpn3.nix:49]: If we order this correct, we should not require restarting openvpn3

**Comment by name-snrl** [comment]
Fixed what i want in #242371

**Pull request closed without merging**